app/src/org/commcare/models/AndroidSessionWrapper.java

```diff
@@ -308,28 +226,30 @@ public static AndroidSessionWrapper mockEasiestRoute(CommCarePlatform platform,
                 if (e.getSessionDataReqs().size() == 1) {
                     //This should fit the bill. Single selection.
                     SessionDatum datum = e.getSessionDataReqs().firstElement();
-
-                    //The only thing we need to know now is whether we have a better option available
-
-                    int countPredicates = CommCareUtil.countPreds(datum.getNodeset());
-
-                    if (wrapper == null) {
-                        //No previous value! Yay.
-                        //Record the degree of specificity of this selection for now (we'll
-                        //actually create the wrapper later
-                        curPredicates = countPredicates;
-                    } else {
-                        //There's already a path to this form. Only keep going 
-                        //if the current choice is less specific
-                        if (countPredicates >= curPredicates) {
-                            continue;
+                    // we only know how to mock a single case selection
+                    if (datum instanceof EntityDatum) {
```

Comment:
Note, these changes are inside of the `mockEasiestRoute` method, which is a brittle helper for a debug feature: manually load form payloads onto device. This feature is rarely used so we don't need to be cautious here.